RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy

```diff
@@ -134,91 +134,88 @@ class BashConverter extends ConfigurationConverter {
         ]) {
             if (cfg.getConfigurationValues().getBoolean(bashFlag[0] as String, bashFlag[1] as Boolean)) text << separator << bashFlag[2] as String
         }
-        return text;
+        return text
     }
 
     StringBuilder appendPathVariables() {
-        StringBuilder text = new StringBuilder();
+        StringBuilder text = new StringBuilder()
 
         //Set a path if necessary.
         text << separator << new BashCommandSet().getSetPathCommand()
-        return text;
+        return text
     }
 
     /**
+     * @param cfg Configuration object. Basically a tree of configuration values that may additionally contain cross-references in the values.
```

Comment:
"Values with unresolved dependencies (i.e. variables that are not declared in the configuration tree) are put at the end. If `returnLeftOverCValuesOnly` is set, only the configuration values referencing undeclared variables are returned." ... or so.

Reply:
Actually, a cleaner interface would be a returning a pair of cvalues with resolved and unresolved references. Then in another small function, the two sets could be combined. The switch would be unnecessary. Furthermore, the name does not indicate, that the function either returns only unresolved or all variables.

Reply:
I am not really sure how to do that now. I changed the comment but I don't have a good idea for the interface. I'd leave it for now and create a new issue.

Reply:
As I said:

```groovy
Map<String, ConfigurationValue> getConfigurationValuesSortedByDependencies(Configuration cfg) {
   Tuple2<Map<String,ConfigurationValue>> sortedAndRest = getConfigurationValuesSortedByDependenciesAndUnresolvable(Configuration cfg)
  return sortedAndRest.first + sortedAndRest.second
}

Tuple2<Map<String,ConfigurationValue>> getConfigurationValuesSortedByDependencies(Configuration cfg) {
   // more or less what you have now
   return new Tuple2(listOfSortedValues, listOfUnsortedValues)
}
```

or so